### ISSUE TITLE
Chrome/Chromium/Firefox support

### ISSUE THREAD

**Issue by dexterleng** [dexterleng/vimac#78]
Chromium and Firefox and other non-native applications do not enable Accessibility by default for performance reasons. You need a way to tell them to enable accessibility, and preferably not keep it on all the time. Ideally, you would only enable it right before querying for elements.

The `AXEnhancedUserInterface` attribute is used by VoiceOver to tell applications that VoiceOver is enabled. Chromium waits for this to be set to true before enabling its Accessibility features. See (https://www.chromium.org/developers/design-documents/accessibility)

Chromium support through `AXEnhancedUserInterface` was removed in PR #62 because the `AXEnhancedUserInterface` attribute when set to 1 on an `AXWindow` breaks window positioning (sluggish too) with window managers like Magnet.

Electron allows you to enable accessibility through the `AXManualAccessibility` attribute, which does not have that window side-effect. Vimac currently sets this attribute to true on frontmost windows.

However, if you would like Vimac to work with Chrome, you can enable Accessibility at all times by enabling it at `chrome://accessibility/`:

![Screenshot 2020-12-06 at 3 17 46 PM](https://user-images.githubusercontent.com/34204380/101274100-37956a00-37d6-11eb-9593-71c00c0cfd27.png)

EDIT:

I have filed issues with Apple, Chromium, and Firefox:

https://developer.apple.com/forums/thread/659755
https://bugs.chromium.org/p/chromium/issues/detail?id=1127357
https://bugzilla.mozilla.org/show_bug.cgi?id=1664992

**Comment by travrob1** [comment]
@dexterleng HI. First off i freaking LOVE your product.  So fantastic. I am a web developer.  Would adding Chromium support give the ability to use the dev tools?
![image](https://user-images.githubusercontent.com/4155964/77794850-30587280-7032-11ea-84d3-7132ad72bd9b.png)

Notice no suggestions on the inspector tools tabs etc...  Thanks

**Comment by chriszrc** [comment]
Similarly, would adding chromium support also mean it would work more for electron apps like Spotify? I so need these shortcuts there-

**Comment by dexterleng** [comment]
@chriszrc yes, at least from the last time I tried. It made windows pretty broken, so keeping it in didn't make sense.

**Comment by dexterleng** [comment]
Spotify is chromium, not electron. Electron support is already present now, some apps just need to update their electron version.

**Comment by dexterleng** [comment]
@travrob1 

Yes it would work with DevTools. I'll file a bug report with Chromium.

<img width="1423" alt="Screenshot 2020-09-11 at 9 25 00 PM" src="https://user-images.githubusercontent.com/34204380/92931014-6c16b580-f475-11ea-9f72-c2f0904fe858.png">

**Comment by dexterleng** [comment]
Just tested `AXEnhancedUserInterface` on Firefox, it works as expected, with the same broken window manager problem.

![Screenshot 2020-09-15 at 4 02 45 PM](https://user-images.githubusercontent.com/34204380/93183228-4e936580-f76d-11ea-8b72-30bbc140e5f4.png)

**Comment by dexterleng** [comment]
@travrob1 You can keep accessibility enabled on Chrome at all times through `chrome://accessibility` at a cost of performance.

See https://www.chromium.org/developers/design-documents/accessibility

**Comment by zhigang1992** [comment]
Is there a way to keep these settings enable for Chrome after relaunching?

Right it reset back to off when relaunch Chrome.

**Comment by dexterleng** [comment]
Hey everyone - if you are interested in getting browser support for Vimac - consider making a PR to these browsers. I do not have the compute power or disk space to build these browsers on my 2017 MacBook Pro.

Filed Issues
https://developer.apple.com/forums/thread/659755
https://bugs.chromium.org/p/chromium/issues/detail?id=1127357
https://bugzilla.mozilla.org/show_bug.cgi?id=1664992

**Comment by dexterleng** [comment]
The Non-Native support Experimental options in v0.3.16 will enable support for Chrome and other non-native browsers:

<img width="647" alt="Screenshot 2021-03-18 at 4 10 49 PM" src="https://user-images.githubusercontent.com/34204380/111593068-83f43b00-8804-11eb-9fd1-41ff993d6655.png">

**Comment by dexterleng** [comment]
I'll close this issue now. If you want Vimac to support Chrome/Other browsers but do not want to deal with this window management bug, consider submitting patches to them to support "AXManualAccessibility" (see https://github.com/dexterleng/vimac/issues/78#issuecomment-765510880)



> Hey everyone - if you are interested in getting browser support for Vimac - consider making a PR to these browsers. I do not have the compute power or disk space to build these browsers on my 2017 MacBook Pro.
> 
> Filed Issues
> https://developer.apple.com/forums/thread/659755
> https://bugs.chromium.org/p/chromium/issues/detail?id=1127357
> https://bugzilla.mozilla.org/show_bug.cgi?id=1664992
